src/protocols/http/hpr_http_roaming_downlink_handler.erl

```diff
@@ -38,25 +37,37 @@ handle(Req, Args) ->
         {error, _} = Err ->
             lager:error("dowlink handle message error ~p", [Err]),
             {500, [], <<"An error occurred">>};
-        {join_accept, {ResponseStream, DownlinkPacket}} ->
-            lager:debug("sending downlink [response_stream: ~p]", [ResponseStream]),
-            ok = send_response(ResponseStream, DownlinkPacket),
-            {200, [], <<"downlink sent: 1">>};
-        {downlink, PayloadResponse, {ResponseStream, DownlinkPacket}, {Endpoint, FlowType}} ->
+        {join_accept, {PubKeyBin, PacketDown}} ->
+            lager:debug("sending downlink [gateway: ~p]", [hpr_utils:gateway_name(PubKeyBin)]),
+            case hpr_packet_router_service:send_packet_down(PubKeyBin, PacketDown) of
+                ok ->
+                    {200, [], <<"downlink sent: 1">>};
+                {error, not_found} ->
+                    {404, [], <<"Not Found">>}
+            end;
+        {downlink, PayloadResponse, {PubKeyBin, PacketDown}, {Endpoint, FlowType}} ->
             lager:debug(
-                "sending downlink [response_stream: ~p] [response: ~s]",
-                [ResponseStream, PayloadResponse]
+                "sending downlink [gateway: ~p] [response: ~s]",
```

Comment:
We should put the gateway in metadata, so its queryable.
And I'm not sure the `PayloadResponse` is safe to print here.